### ISSUE TITLE
SSR | State from Server not following over to Client

### ISSUE THREAD

**Issue by LucasLundJensen** [pulse-framework/pulse#162]
This is a question, and not a report / issue.
The SSR feature of NextJS with Pulse doesn't have a lot of examples, so I'm asking the question here.

I've set some state inside getServerSideProps and used preserveServerState on my const data = {props: {}}.

## Expected Behavior

I'd expect the state to be 'magically' set on the client (through preserveServerState), from the state that was set on the server.

## Current Behavior

The state on the client isn't set, but I do get my PULSE_DATA with the state into the page's props, but am I supposed to set my client state from the result of those props?
It seems counter intuitive, to set state twice... But I'm sure it's just me not figuring it out correctly.

## Extra Information

Am I doing something wrong, or is my result the expected behavior, and am I just supposed to set state on client, from the props returned from getServerSideProps?

## Environment Information
- Operating System: `Windows/10 && Mac OS Big Sur/11.1`
- Browser Version (If applicable): `Chrome/88.0.4324.192`
- Pulse Version: `3.5.2`
- Framework/Framework Version (If applicable): `NextJS - 3.5.4`

**Comment by LucasLundJensen** [comment]
It also seems like that when the state changes on server side, the new values don't get thrown with back to PULSE_DATA, it's the old values that I got on page load, which are still in the PULSE_DATA object.

Here is my code, so you get an idea of what I'm doing.

Page:
`
export async function getServerSideProps(context) {
  const data = { props: {} };
  await authorize(context.req, context.res);
  return preserveServerState(data);
}
`

Authorize function (Server Side):
`
export async function authorize(req: NextApiRequest, res: NextApiResponse) {
  const refreshToken = getRefreshTokenFromReq(req);
  if (!refreshToken) return null;

  const response = await fetch("http://localhost:3001/api/auth/refresh", {
    credentials: "include",
    headers: {
      cookie: serialize("refreshToken", refreshToken, {
        path: '/'
      }),
    },
  });
  const json = await response.json();
  
  if (json.status === "MissingCredentials") {
    return { status: json.status };
  } else {
    core.authentication.state.TOKEN.set(json.accessToken);
    core.authentication.state.AUTHENTICATED.set(true);
    await core.user.actions.getUserInformation();
  }
}
`

**Comment by itsRems** [comment]
2nd part is a known issue with .reset() not "running" if the default value is not set, but then if it's undefined, pulse will think it's not set.
@jamiepine @TwanLuttik @hadenpf thoughts about removing the check and allowing .reset to set the state to undefined ?

**Comment by LucasLundJensen** [comment]
I've tried to follow the advice given by @itsRems both on this issue and on his stream.
But I now get the following error:

`react-dom.development.js?61bb:67 Warning: Text content did not match. Server: "2" Client: "1" which of course still is an issue with the state not following over to the client..
None the less, it's an improvement as to before, where I didn't get any error message, and the scenario was a little differnet.

I've made a repo with the issue.
The repository is: https://github.com/LucasLundJensen/next-issue

`Counter basically just goes up by 1 on every refresh`
